### PR TITLE
refactor(scan): detect QR codes in `ballot-interpreter-vx`

### PR DESCRIPTION
**Pull request by eventualbuddha**
## Overview

Refs https://github.com/votingworks/vxsuite/issues/1512

The way we look for QR codes is pretty specific to the `vx` ballot, and doesn't really generalize. At that, QR codes on non-BMD ballots is already pretty edge-case. Because of this it makes more sense for this code to live in `ballot-interpreter-vx`.

Doing this required moving some tests and fixtures, as well as removing the QR code libraries as direct dependencies of `services/scan`. For now we're still calling `detectQrCode` from `services/scan` since we're using worker processes to look for both front/back QR codes at once. The next step will be to move this optimization into `ballot-interpreter-vx` too.

## Demo Video or Screenshot
n/a

## Testing Plan 
Automated testing.

## Checklist
- [x] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.
- [x] I have added JSDoc comments to any newly introduced exports

### CODE REVIEW

File: libs/ballot-interpreter-vx/src/interpreter/basics.test.ts [hunk 1]
```diff
@@ -1,36 +1,10 @@
...
 import { fail } from 'assert';
+import { Interpreter } from '.';
 import * as choctaw2020LegalSize from '../../test/fixtures/choctaw-county-2020-general-election';
 import * as oaklawn from '../../test/fixtures/election-4e31cb17d8-ballot-style-77-precinct-oaklawn-branch-library';
-import { Interpreter } from '.';
-import { DetectQrCodeResult } from '../types';
 
 jest.setTimeout(10000);
 
-test('custom QR code reader', async () => {
```
Comment (eventualbuddha) [inline libs/ballot-interpreter-vx/src/interpreter/basics.test.ts:10]: This test was no longer relevant since the functionality was removed.

File: libs/ballot-interpreter-vx/src/utils/qrcode/with_cropping.ts [hunk 2]
```diff
@@ -1,56 +0,0 @@
-import { Rect } from '@votingworks/types';
-import { DetectQrCode, DetectQrCodeResult } from '../../types';
-import { crop } from '../crop';
-import { flipRectVerticalAndHorizontal } from '../geometry';
-
-/**
- * Makes a new QR code detector from an existing one that works by cropping the
- * bottom-right and top-left corners of the image before passing it on.
- */
-export function withCropping(
```
Comment (eventualbuddha) [inline libs/ballot-interpreter-vx/src/utils/qrcode/with_cropping.ts:10]: This functionality was all duplicated in `services/scan`, but in a totally different way. Fun!

File: libs/ballot-interpreter-vx/src/interpreter/two_column_template.test.ts [hunk 3]
```diff
@@ -8,21 +8,23 @@ test('interpret two-column template', async () => {
   const interpreter = new Interpreter({ election });
 
   {
-    const imageData = await choctawMock2020.blankPage1.imageData();
     const template = await interpreter.addTemplate(
-      await interpreter.interpretTemplate(imageData)
+      await interpreter.interpretTemplate(
+        await choctawMock2020.blankPage1.imageData(),
+        // provide the metadata because the QR code uses raw binary, not base64
+        // binary. this causes qrdetect (zbar) to read it incorrectly
```
[the remark this answers is not in the document]
Comment (jonahkagan) [reply libs/ballot-interpreter-vx/src/interpreter/two_column_template.test.ts:15]: does it merit creating a task?

**Review by jonahkagan — APPROVED** [review]

File: libs/ballot-interpreter-vx/src/interpreter/two_column_template.test.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (eventualbuddha) [reply libs/ballot-interpreter-vx/src/interpreter/two_column_template.test.ts:15]: https://github.com/votingworks/vxsuite/issues/1524
